projects/igniteui-angular/src/lib/grids/grid/groupby-row.component.ts

```diff
@@ -8,26 +8,42 @@ import {
     Input,
     ViewChild,
     TemplateRef,
+    OnDestroy
 } from '@angular/core';
 import { IGroupByRecord } from '../../data-operations/groupby-record.interface';
 import { DataType } from '../../data-operations/data-util';
 import { GridBaseAPIService } from '../api.service';
 import { IgxGridBaseDirective } from '../grid-base.directive';
 import { IgxGridSelectionService, ISelectionNode } from '../selection/selection.service';
 import { GridType } from '../common/grid.interface';
+import { IgxFilteringService } from '../filtering/grid-filtering.service';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
+import { GridSelectionMode } from '../common/enums';
+import { IgxGridRowComponent } from './grid-row.component';
 
 @Component({
     changeDetection: ChangeDetectionStrategy.OnPush,
     preserveWhitespaces: false,
     selector: 'igx-grid-groupby-row',
     templateUrl: './groupby-row.component.html'
 })
-export class IgxGridGroupByRowComponent {
+export class IgxGridGroupByRowComponent implements OnDestroy {
 
     constructor(public gridAPI: GridBaseAPIService<IgxGridBaseDirective & GridType>,
-        private gridSelection: IgxGridSelectionService,
+        public gridSelection: IgxGridSelectionService,
         public element: ElementRef,
-        public cdr: ChangeDetectorRef) { }
+        public cdr: ChangeDetectorRef,
+        public filteringService: IgxFilteringService) {
+        this.gridSelection.selectedRowsChange.pipe(takeUntil(this.destroy$)).subscribe(() => {
+            this.cdr.markForCheck();
```

Comment:
There is no need to subscribe to an observable every time a row is selected, since this will be triggered for every group row present in the view and adding additional observable to the service anyway is not needed.

You can use the already available detection cycle and implement `ngDoCheck`, where you can directly call `this.cdr.markForCheck();` or have properties that contain latest values for allRowsInGroupSelected or anyRowInGroupSelected or similar and then call `this.cdr.markForCheck();` if they are changed.

Reply:
Totally depends on if the check can be made on something super quick, because `ngDoCheck` is quite chatty and if we have to do a lot of collection iterations and checks it might negate the any possible gains of not triggering all group rows at once.
Will have to test that (time allowing)